application/helpers/admin/import_helper.php

```diff
@@ -1272,6 +1268,26 @@ function XMLImportSurvey($sFullFilePath, $sXMLdata = null, $sNewSurveyName = nul
         if (isset($insertdata['tokenlength']) && $insertdata['tokenlength'] > 35) {
             $insertdata['tokenlength'] = 35;
         }
+
+        // Fix survey group when not copying
+        if (!$isCopying) {
+            // Set to 1 by default
+            $insertdata['gsid'] = 1;
+            // If the xml includes survey group details, try to find the group by name.
+            if (!empty($xml->surveys_groups->rows->row[0]->name)) {
+                $surveyGroupName = (string) $xml->surveys_groups->rows->row[0]->name;
+                $surveyGroupPermissionCriteria = SurveysGroups::getPermissionCriteria();
+                $surveyGroup = SurveysGroups::model()->findByAttributes(["name" => $surveyGroupName], $surveyGroupPermissionCriteria);
+                // If a survey group is found with the specified name, assign it to the survey.
```

Comment:
This disallow (currently) Permssion event + must check Permission to "import" survey in this group

It's not because you have the right to view SurveyGroup you have the right to import survey in this group

Then must be
```
$surveyGroup = SurveysGroups::model()->findByAttributes(["name" => $surveyGroupName]);
if(!empty($surveyGroup) && SurveysInGroup::hasPermission('survey','import) {
```

Reply:
You are right. Got confused.

Reply:
Still an issue about `SurveysInGroup::hasPermission('survey','import')` this need a fix in hasPermission …

Reply:
Actually, this `getPermissionCriteria` is the same method used by the create survey screen when listing available groups.
That's why we use it. As to try to use the same.

I think that if there is an issue, it is a more general one.

Also, you mention
> It's not because you have the right to view SurveyGroup you have the right to import survey in this group

But then, you also mention this on 18530
> User can add survey in group only if he can see the group. There is no 'Create' for this permission.

I believe there is some kind of problem on permission management for the surveygroups, but for this fix, we are kind of copying what already is available

Reply:
> I believe there is some kind of problem on permission management for the surveygroups, but for this fix, we are kind of copying what already is available. So maybe we should just treat that on chat, another ticket or somehting else?

2 fix needed ? 
see https://bugs.limesurvey.org/view.php?id=18530

Reply:
I would just leave this as it is now, which is the same as the create survey.
And then, after 18530, if we need, we update it.
As for simplifying the process and not try to solve all at once

Reply:
Thought that. Was the seed of this approach. But seems better to just update getPermissionCriteria.
Why do I need to fetch all surveygroups if only interested in one?

Reply:
> Thought that. Was the seed of this approach. But seems better to just update getPermissionCriteria.

getPermissionCriteria exist on other model, i think i made an error : it need to be in Trait
See related issue : https://bugs.limesurvey.org/view.php?id=16459

And because currently ; it's the only way to get valid list of gsid

> Why do I need to fetch all surveygroups if only interested in one?

because it's already public, see https://github.com/LimeSurvey/LimeSurvey/pull/2752/files?diff=unified&w=0#diff-c2e2b37076b5bae18bcd942f811752f5be549ebcec44fd866ffbba17662ca35eR332
(related issue https://bugs.limesurvey.org/view.php?id=18530)
Here : i add an optional gsid on the list to get only the needed one.

Reply:
PS : maybe just try `SurveysInGroup::model()->findByPK($gsid)->hasPermission('surveys', 'import')` ?

Since the issue is related to this ?

Reply:
> Here : i add an optional gsid on the list to get only the needed one.

Actually, we thought about that too. But then the name of the method will not make much sense.
Get a list of survey groups with only a single item?

> getPermissionCriteria exist on other model, i think i made an error : it need to be in Trait
We can create an alias to expose the method.
When you fix the method, it shouldn't impact.

Reply:
> We can create an alias to expose the method.
> When you fix the method, it shouldn't impact.

Here : the fix must use `SurveysInGroup::model()->findByPK($gsid)->hasPermission('surveys', 'import')`  right ?

Except current permission system is broken and need to be fixed.
https://github.com/LimeSurvey/LimeSurvey/pull/2752

Since it's another issue : maybe leave with just `SurveysInGroup::model()->findByPK($gsid)->hasPermission('surveys', 'import')` ?